opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/__init__.py

```diff
@@ -14,72 +14,74 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import argparse
+from argparse import REMAINDER, ArgumentParser
 from logging import getLogger
 from os import environ, execl, getcwd
 from os.path import abspath, dirname, pathsep
+from re import sub
 from shutil import which
 
-from opentelemetry.environment_variables import (
-    OTEL_PYTHON_ID_GENERATOR,
-    OTEL_TRACES_EXPORTER,
-)
+from pkg_resources import iter_entry_points
 
-logger = getLogger(__file__)
+_logger = getLogger(__file__)
 
 
-def parse_args():
-    parser = argparse.ArgumentParser(
+def run() -> None:
+
+    parser = ArgumentParser(
         description="""
         opentelemetry-instrument automatically instruments a Python
         program and its dependencies and then runs the program.
-        """
+        """,
+        epilog="""
+        Optional arguments (except for --help) for opentelemetry-instrument
+        directly correspond with OpenTelemetry environment variables. The
+        corresponding optional argument is formed by removing the OTEL_ or
+        OTEL_PYTHON_ prefix from the environment variable and lower casing the
+        rest. For example, the optional argument --attribute_value_length_limit
+        corresponds with the environment variable
+        OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT.
```

Comment:
I feel we shouldn't call out that these CLI args correspond to env vars. It'll create an expectation that all env vars must be supported here as CLI args which may or may not be viable for us to do.

Reply:
Do you foresee an scenario where supporting environment variables in this fashion wouldn't be possible?

Reply:
Not particularly but in general I think it'd be safer or more flexible if we just document the feature instead of documenting an implementation detail.

> these are CLI args you can use to configure OpenTelemetry with the opentelemetry-instrument command 

instead of saying

>  These cli args are sources from environment variables of some specific packages or packages that implement a given entry point.

May be in future we'll decide to move the source of truth somewhere else instead of env vars. Let's say we add support for config files and the files have a schema. May be that schema will become the source of truth which will be sources by both env vars and cli args.

That said, not a big deal or blocker for me. I'm fine with merging as is.

Reply:
I feel like this is useful information, if we change the source of arguments we can update the documentation later.